### PR TITLE
Add paint metrics to Performance Timeline API

### LINKED ISSUE #18111: Add paint metrics to Performance Timeline API
WIP at https://github.com/ferjm/servo/commit/2042941a3edf0cabc22aee4add1049e61e892563

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #18111
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18155)

### CODE REVIEW

File: components/script/dom/performancepainttiming.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            entry: PerformanceEntry::new_inherited(name,
+                                                   DOMString::from("paint"),
+                                                   start_time,
+                                                   0.)
+        }
+    }
+
+    #[allow(unrooted_must_root)]
+    pub fn new(global: &GlobalScope,
+           name: DOMString,
```
Comment (jdm) [inline components/script/dom/performancepainttiming.rs]: Let's accept the enum here and convert it into a string in `new_inherited`.

File: tests/wpt/mozilla/tests/mozilla/paint_timing.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>Paint Timing API</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({
+  explicit_done: true
+});
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/paint_timing.html]: Let's do `var t = async_test("Performance entries observer");` instead.

File: tests/wpt/mozilla/tests/mozilla/paint_timing.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+var observer = new PerformanceObserver(list => {
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/paint_timing.html]: Let's do `new PerformanceObserver(t.step_func_done(list => {`

File: tests/wpt/mozilla/tests/mozilla/paint_timing.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  list.getEntries().forEach((entry, i) => {
+    var name = i == 0 ? "first-paint" : "first-contentful-paint";
+    assert_equals(entry.name, name, "Name is " + name);
+    assert_equals(entry.entryType, "paint", "Entry type is paint");
+    assert_true(entry.startTime > 0, "Start time is > 0");
+    assert_equals(entry.duration, 0, "Duration is 0");
+  });
+  done();
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/paint_timing.html]: We don't need the explicit `done()` any more.

**Review by jdm — CHANGES_REQUESTED** [review]
The implementation looks fine to me.

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /_mozilla/mozilla/interfaces.html:
  │ FAIL [expected PASS] Interfaces exposed on the window
  │   → assert_true: If this is failing: DANGER, are you sure you want to expose the new interface PerformancePaintTiming to all webpages as a property on the global? Do not make a change to this file without review from jdm or Ms2ger for that specific change! expected true got false
  │ 
  │ test_interfaces/<@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:73:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  │ test_interfaces@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:2:3
  └ @http://web-platform.test:8000/_mozilla/mozilla/interfaces.html:13:1

  ▶ Unexpected subtest result in /_mozilla/mozilla/interfaces.worker.html:
  │ FAIL [expected PASS] Untitled
  │   → assert_true: If this is failing: DANGER, are you sure you want to expose the new interface PerformancePaintTiming to all webpages as a property on the global? Do not make a change to this file without review from jdm or Ms2ger for that specific change! expected true got false
  │ 
  │ test_interfaces/<@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:73:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  │ test_interfaces@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:2:3
  └ @http://web-platform.test:8000/_mozilla/mozilla/interfaces.worker.js:9:1
```
Go ahead.

**Comment by ferjm** [comment]
Hmm... I used to be able to trigger retries :\

**Comment by jdm** [comment]
@bors-servo: delegate+
This needs to update the manifest before it can merge.

**Pull request merged**